From 1b79eecd9c5c71237ca8960319dc4d858895220d Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Mon, 3 Nov 2025 20:36:50 +0100
Subject: [PATCH] git: update to v2.52.0

Update to git version v2.52.0, this requires changes for these
upstream commits:

* 9f6dfe43c8a55b833ae16486bcafe29b543461f9
  string-list: align string_list_split() with its _in_place() counterpart

* 78237ea53d6546aeab7adb2c7547a1177311ccde
  packfile: split up responsibilities of `reprepare_packed_git()`

* 751808b2a18acba76b824aed4d8b7442bd7f5fca
  packfile: refactor `get_packed_git()` to work on packfile store

Signed-off-by: Christian Hesse <mail@eworm.de>
---
 Makefile   | 2 +-
 shared.c   | 2 +-
 ui-clone.c | 4 ++--
 4 files changed, 5 insertions(+), 5 deletions(-)

diff --git Makefile Makefile
index 08b4c90..c376d2f 100644
--- Makefile
+++ Makefile
@@ -14,7 +14,7 @@ htmldir = $(docdir)
 pdfdir = $(docdir)
 mandir = $(prefix)/share/man
 SHA1_HEADER = <openssl/sha.h>
-GIT_VER = 2.51.1
+GIT_VER = 2.52.0
 GIT_URL = https://www.kernel.org/pub/software/scm/git/git-$(GIT_VER).tar.xz
 INSTALL = install
 COPYTREE = cp -r
diff --git shared.c shared.c
index 3838364..401cf86 100644
--- shared.c
+++ shared.c
@@ -395,7 +395,7 @@ int cgit_parse_snapshots_mask(const char *str)
 	if (strcmp(str, "all") == 0)
 		return INT_MAX;
 
-	string_list_split(&tokens, str, ' ', -1);
+	string_list_split(&tokens, str, " ", -1);
 	string_list_remove_empty_items(&tokens, 0);
 
 	for_each_string_list_item(item, &tokens) {
diff --git ui-clone.c ui-clone.c
index 0fadf24..9445454 100644
--- ui-clone.c
+++ ui-clone.c
@@ -40,8 +40,8 @@ static void print_pack_info(void)
 	ctx.page.mimetype = "text/plain";
 	ctx.page.filename = "objects/info/packs";
 	cgit_print_http_headers();
-	reprepare_packed_git(the_repository);
-	for (pack = get_packed_git(the_repository); pack; pack = pack->next) {
+	odb_reprepare(the_repository->objects);
+	for (pack = packfile_store_get_packs(the_repository->objects->packfiles); pack; pack = pack->next) {
 		if (pack->pack_local) {
 			offset = strrchr(pack->pack_name, '/');
 			if (offset && offset[1] != '\0')
-- 
2.51.2

